staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -51,6 +51,11 @@ func NewEventFromRequest(req *http.Request, level auditinternal.Level, attribs a
 	// prefer the id from the headers. If not available, create a new one.
 	// TODO(audit): do we want to forbid the header for non-front-proxy users?
 	ids := req.Header[auditinternal.HeaderAuditID]
+
+	// golang canonicalizes the header name when adding values to the header map
+	// see https://golang.org/src/net/http/request.go#L150 for reference
+	ids = append(ids, req.Header[http.CanonicalHeaderKey(auditinternal.HeaderAuditID)]...)
```

Comment:
I wonder, shouldn't we switch `auditinternal.HeaderAuditID` to the canonical form?